api/games/create.js

```diff
@@ -446,12 +446,40 @@ export const createGameFromLobby = gameLobby => {
   const { onRoundStart, onGameStart, onStageStart } = config;
   if ((onGameStart || onRoundStart || onStageStart) && firstRoundId) {
     const game = Games.findOne(gameId);
-    const players = Players.find({
-      _id: { $in: params.playerIds }
-    }).fetch();
-    game.treatment = treatment.factorsObject();
-    game.players = players;
-    game.rounds = Rounds.find({ gameId }).fetch();
+    let gameTreatment = null,
+      gamePlayers = null,
+      gameRounds = null;
+
+    Object.defineProperties(game, {
+      treatment: {
+        get() {
+          if (gameTreatment) {
+            return gameTreatment;
+          }
+          gameTreatment = treatment.factorsObject();
+          return gameTreatment;
+        }
+      },
+      players: {
+        get() {
+          if (gamePlayers) {
+            return gamePlayers;
+          }
+          gamePlayers = game.getPlayers();
+          return gamePlayers;
+        }
+      },
+      rounds: {
+        get() {
+          if (gameRounds) {
+            return gameRounds;
+          }
+          gameRounds = game.getRounds();
+          return gameRounds;
+        }
+      }
+    });
+
     game.rounds.forEach(round => {
       round.stages = Stages.find({ roundId: round._id }).fetch();
     });
```

Comment:
All that work to lazy load those object is lost if you're going to call those objects in the next line. Both `game.rounds` and `game.players` are called immediately, so we're not improving anything here.

To actually improve things, you needs to nest all this further loading of object into the get() functions.


```suggestion
```